src/common/config.h

```diff
@@ -338,6 +339,12 @@ template<typename T> T md_config_t::get_val(const std::string &key) const {
   return boost::apply_visitor(gtv, generic_val);
 }
 
+template<typename T> T md_config_t::_get_val(const std::string &key) const {
+  Option::value_t generic_val = this->get_val_generic(key);
+  get_typed_value_visitor<T> gtv;
+  return boost::apply_visitor(gtv, generic_val);
```

Comment:
it's identical to 
```
template<typename T> T md_config_t::get_val(const std::string &key) const
```.

why shall we use it instead of `md_config_t::get_val()` ?

Reply:
Thanks for catching this. It was SUPPOSED to _get_val() instead of get_value_generic since _get_val desn't take a lock. I'll push an update in a moment.

Reply:
@adamemerson it's not changed. am i missing something?